pr/game.py

```diff
@@ -1,150 +1,184 @@
 import logger
 
-board = []
-next_char = 'o'
-players = {'x': '', 'o': ''}
-is_first_player = True
-
-
-def prepare():
-    global board
-    set_players_name()
-    board_size = get_meaning('Input board size more than 2: ', 2)
-    board = create_board(board_size)
-    logger.show_board(board)
-
-
-def set_players_name():
-    global players
-    print('Hi! Let\'s play!')
-    players['x'] = input('Input first player name (it will play with x): ')
-    players['o'] = input('Input second player name (it will play with o): ')
-
-
-def get_meaning(message, down_board, up_board=20):
-    is_incorrect_input = True
-    player_input = 0
-    while is_incorrect_input:
-        try:
-            player_input = int(input(message))
-        except ValueError:
-            print('Incorrect input!')
-        else:
-            if player_input < down_board or player_input > up_board:
-                print('Incorrect meaning!')
-            else:
-                is_incorrect_input = False
 
-    return player_input
+class Game:
+    __board = []
+    __next_char = ''
+    __players = {'x': '', 'o': ''}
+    __is_first_player = True
+    __cell_counter = 0
+    __board_size = len(__board)
+    log = logger.Logger
+
+    def __init__(self):
+        self.__board = []
+        self.__next_char = 'o'
+        self.__players = {'x': '', 'o': ''}
+        self.__is_first_player = True
+        self.__cell_counter = 0
+        self.__board_size = len(self.__board)
+
+    def play_game(self):
+        self.log.log_to_file('Start of game!')
+        next_game = True
+
+        while next_game:
+            self.__prepare_game()
+            self.__round_game()
+            answer = input('New game? (Y/N) ')
+            if answer != 'Y' and answer != 'y':
+                next_game = False
+
+        self.log.log_to_file('Game is over!')
+
+    def __round_game(self):
+        self.log.log_to_file('New round!')
+        next_round = True
+
+        while next_round:
+            self.__play_round()
+            answer = input('New round? (Y/N) ')
+            if answer != 'Y' and answer != 'y':
+                next_round = False
+            else:
+                self.__prepare_round()
+
+        self.log.log_to_file('Round is over!')
+
+    def __play_round(self):
+        is_round_over = False
+
+        while not is_round_over:
```

Comment:
Same like code line 25. Can be simplified

Reply:
Fixed